croncat/src/system.rs

```diff
@@ -40,10 +48,15 @@ pub async fn run(env: Env, shutdown_tx: ShutdownTx, shutdown_rx: ShutdownRx) ->
     // Account status checks
     let account_status_check_shutdown_rx = shutdown_rx.clone();
     let account_status_check_block_stream_rx = block_stream_rx.clone();
+    let block_status = Arc::new(Mutex::new(initial_status));
```

Comment:
would love to hear about this `Arc` and `Mutex` in our next call)) I've read about it but glad we're getting to use it in production